packages/rpc-provider/src/ws/Provider.ts

```diff
@@ -124,7 +131,8 @@ export default class WsProvider implements WSProviderInterface {
     try {
       const WS = await getWSClass();
 
-      this.#websocket = new WS(this.#endpoint);
+      this.#websocket = new WS(this.#endpoint[this.#endpointNext]);
+      this.#endpointNext = (this.#endpointNext + 1) % this.#endpoint.length;
```

Comment:
```suggestion
      this.#endpointNext = (this.#endpointNext + 1) % this.#endpoints.length;
```

(Just aligning with the `s/endpoint/endpoints/` rename above)

Reply:
Would also suggest moving this below all `this.#websocket` stuff (just keeping the grouping together)